web/src/engine/attachment/src/pageContextAttachment.ts

```diff
@@ -0,0 +1,1130 @@
+    if(this.isAttached(Pelem)) {
+      const intendedInputMode = Pelem._kmwAttachment.inputMode;
+
+      this.disableInputModeObserver();
+      Pelem.inputMode = intendedInputMode;
+      this.enableInputModeObserver();
+    }
+  }
+
+  //#endregion Prime for removal! ----------------------
+
+  /**
+   * The core method for a MutationObserver that checks for changes to the `.inputMode` property
+   * of controls that KMW is attached to in touch mode.
+   *
+   * In touch mode, KMW requires that their `.inputMode` property be set to 'none' in order
+   * to hide the device's default OSK.  That said, we should still aim to honor the setting
+   * and restore it if and when detachment occurs.  Should we ever support intents, we'll want
+   * to utilize the incoming value for use with that feature too.
+   */
+  private _InputModeObserverCore = (mutations: MutationRecord[]) => {
+    // Prevent infinite recursion from any changes / updates made within the observation handler.
+    this.disableInputModeObserver();
+    try {
+      for(const mutation of mutations) {
+        const target = mutation.target as HTMLElement;
+        if(!this.isAttached(target)) {
+          continue;
+        }
+
+        target._kmwAttachment.inputMode = target.inputMode;
+
+        if(this.device.touchable) {
+          target.inputMode = 'none';
+        }
+      }
+    } finally {
+      this.enableInputModeObserver();
+    }
+  };
+
+  /**
+     * Function     _AttachToIframe
+     * Scope        Private
+     * @param       {Element}      Pelem       IFrame to which KMW will be attached
+     * Description  Attaches KeymanWeb to IFrame
+     */
+  _AttachToIframe(Pelem: HTMLIFrameElement) {
+    try {
+      const Lelem=Pelem.contentWindow.document;
+      /* editable Iframe */
+      if(Lelem) {
+        if(Lelem.designMode.toLowerCase() == 'on') {
+          // Set up a reference alias; the internal document will need the same attachment info!
+          this.setupElementAttachment(Pelem);
+          Lelem.body._kmwAttachment = Pelem._kmwAttachment;
+
+          this._inputList.push(Pelem);
+          this.emit('enabled', Pelem);
+        } else {
+          // If already attached, do not attempt to attach again.
+          if(this.embeddedPageContexts.filter((context) => context.document == Lelem).length == 0) {
+            // Lelem is the IFrame's internal document; set 'er up!
+            let embeddedPageAttachment = new PageContextAttachment(Lelem, {
+              ...this.options,
+              owner: Pelem
+            });
+
+            this.embeddedPageContexts.push(embeddedPageAttachment);
+            // Forward any attached elements from the embedded page as if we attached them directly.
+            embeddedPageAttachment.on('enabled',  (elem) => this.emit('enabled',  elem));
+            embeddedPageAttachment.on('disabled', (elem) => this.emit('disabled', elem));
+
+            embeddedPageAttachment.install(this.manualAttach);
+          }
```

Comment:
Obviously, this point is a bit divergent from the original at first glance.  Before... we'd just call `_SetupDocument` again.  Why the change here?

The design I landed on for handling attachment works best when it can run from the perspective of a page's document.  The organizational pattern is much clearer this way.  As it turns out, given this decision, the simplest way to handle IFrame-internal elements... turned out to be spawning a child instance for the child's document.  A little bit of event-forwarding later, and this aspect is invisible from the outside.

I'm pretty sure this was the mental model behind the original design as well... it just didn't encompass _quite_ all of KMW's event listeners.

Worth noting, from the description:  
> During the modularization process, I came to notice that there are a number of top-page-only handlers that really should be set on every page if possible, not just the top-level page. However, as KMW 16.0 and before have maintained a distinction... the current design will continue to do so. Note the optional owner field for the object expected by the constructor - this is used within related conditionals to prevent 'owned' documents from setting top-level-only handlers in place and maintain the pre-existing distinction.

This change gives us a very direct, straightforward way to set those currently "top-level-only" event listeners within child iframe documents as well, should it prove necessary for issue resolution in the future.  As our current focus is on maintaining existing behavior, I've kept that turned off for now.